service/service.go

```diff
@@ -62,6 +64,10 @@ func newService(set *settings) (*service, error) {
 		telemetryInitializer: set.telemetry,
 	}
 
+	if set.Config.Service.Telemetry.Traces.Propagation.Enabled {
```

Comment:
If this is a feature gate instead we can eliminate this once we're comfortable having propagation enabled by default.  The `none` "propagator" still provides a way for users to prevent propagation.

Reply:
@Aneurysm9 just for the sake of my understanding - some clarifications (sorry never dealt with feature gate module before) 
1. By feature gate are you referring to https://github.com/open-telemetry/opentelemetry-collector/tree/main/service/featuregate? 
2. If that is what you are referring to, are you suggesting that we enable the context propagation by default (assuming this is what we want to do eventually) behind a feature gate flag and users can choose to set the OTEL_PROPAGATOR to `none` should they choose to disable the propagation. Thus eliminating need to have any configuration defined completely? 
3. Is there a good example on some features behind the feature gates?

Reply:
The feature gate sounds good to me and allows us to commit less strongly to this, but I don't think `OTEL_PROPAGATOR` should be the only way to configure this when this is stable (to be honest, I think it's bad that it can be configured at all through an environment variable that we don't control, but I guess we can't avoid that unless the OTel Go SDK changes). If `enabled` is not appropriate, we should have a configuration option equivalent to `OTEL_PROPAGATOR` on the YAML.

Reply:
> we should have a configuration option equivalent to `OTEL_PROPAGATOR` on the YAML.

Agreed.  I think this could be accomplished currently by using `os.SetEnv` to set the value of `OTEL_PROPAGATOR` if it isn't already set.  I'll propose that an option be added to the autoprop interface to allow setting a default propagator by ID rather than just by providing an implementation.

> (to be honest, I think it's bad that it can be configured at all through an environment variable that we don't control, but I guess we can't avoid that unless the OTel Go library changes)

This is not going to change.  OTel in general has made the decision that SDKs should be configurable through environment variables and this is in line with that policy.  I think there is significant benefit to operators that may not be able to alter configuration in other ways but do have the ability to control the execution environment.  Ultimately, it would be little different from this configuration:

```yaml
service:
  telemetry:
    trace:
      propagator: ${OTEL_PROPAGATOR}
```

Reply:
>>>  I'll propose that an option be added to the autoprop interface to allow setting a default propagator by ID rather than just by providing an implementation.

+1